Fail when using multiple data paths

rjernst opened the pull request:
This commit converts the deprecation messages for multiple data paths
into errors. It effectively removes support for multiple data paths.

relates #71205

elasticmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

jasontedor reviewed (approved):
LGTM. When CI is happy, I'm happy.

rjernst committed:
This commit converts the deprecation messages for multiple data paths
into errors. It effectively removes support for multiple data paths.

relates #71205

rjernst committed:
checkstyle
